app/src/main/java/com/asfoundation/wallet/ui/iab/InAppPurchaseInteractor.java

```diff
@@ -243,34 +243,27 @@ private Payment mapToBdsPayment(Payment transaction, Purchase purchase) {
         .flatMap(paymentMethods -> getAvailablePaymentMethods(transaction, paymentMethods).flatMap(
             availablePaymentMethods -> Observable.fromIterable(paymentMethods)
                 .map(paymentMethod -> mapPaymentMethods(paymentMethod, availablePaymentMethods))
-                .flatMap(paymentMethod -> getDisableReason(paymentMethod, transaction))
+                .flatMap(paymentMethod -> retrieveDisableReason(paymentMethod, transaction))
                 .toList())
             .map(this::removePaymentMethods))
         .map(this::swapDisabledPositions);
   }
 
-  private Observable<PaymentMethod> getDisableReason(PaymentMethod paymentMethod,
+  private Observable<PaymentMethod> retrieveDisableReason(PaymentMethod paymentMethod,
       TransactionBuilder transaction) {
     if (!paymentMethod.isEnabled()) {
       if (paymentMethod.getId()
           .equals(CREDITS_ID)) {
-        return Observable.just(R.string.purchase_appcoins_credits_noavailable_body)
-            .map(reason -> {
-              paymentMethod.setDisabledReason(reason);
-              return paymentMethod;
-            });
+        paymentMethod.setDisabledReason(R.string.purchase_appcoins_credits_noavailable_body);
       } else if (paymentMethod.getId()
           .equals(APPC_ID)) {
         return getAppcDisableReason(transaction).map(reason -> {
```

Comment:
I think we should keep it nullable instead of -1, it's easier. We can on the map() check if it's -1 and if it is we  don't set the reason